lib/prawn/images/png.rb

```diff
@@ -108,7 +108,7 @@ def colors
       #
       def split_alpha_channel!
         if alpha_channel?
-          if color_type == 3
+          if color_type == 3 && !@transparency.empty?
```

Comment:
`@transparency.any?` seem a bit shorter and simpler.

But that's just nitpicking.